src/Tasks/AssemblyDependency/TaskExecutionContextExtension.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Tasks
+{
+    public static class TaskExecutionContextExtension
```

Comment:
Do not need public here?

Reply:
I don't think so, at least as long as these methods are fixed rather than user-manipulatable. Should this explicitly extend TaskExecutionContext?

Reply:
yes, there was no need for "?" mark in my comment, actually, this has to be private. That is a reminder for myself to fix this together with the other changes.